### PR TITLE
Update versions

### LINKED ISSUE #267: pytest >= 8.2 is not yet supported (not only tests fail)
See https://github.com/pytest-dev/pytest-rerunfailures/actions/runs/8409152908/job/23026172136

This ~~can be a temporary problem or a~~ is a hint that we have to change something here to make it compatible again.

### PR DESCRIPTION
**Pull request by icemac**
- Drop support for Python 3.8
- Drop support for pytest < 7.4
- Add support for Python 3.13
- Add support for pytest 8.2 and 8.3.

Fixes #267

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -46,7 +47,7 @@ classifiers = [
 ]
 dependencies = [
   "packaging>=17.1",
-  "pytest>=7.2",
+  "pytest>=7.4",
```
Comment (asottile-sentry) [inline pyproject.toml:50]:
```suggestion
  "pytest>=7.4,!=8.2.2",
```

as per https://github.com/pytest-dev/pytest-rerunfailures/issues/267#issuecomment-2457891221

File: pyproject.toml [hunk 2]
```diff
@@ -38,6 +38,7 @@ classifiers = [
   "Programming Language :: Python :: 3.10",
   "Programming Language :: Python :: 3.11",
   "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
```
Comment (hugovk) [inline pyproject.toml:41]: Also remove the 3.8 classifier above and bump `requires-python`.

**Review by hugovk — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (icemac) [reply pyproject.toml:40]: Good catch!

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**